WebApiOrderFood/Program.cs

```diff
@@ -28,6 +29,9 @@
 builder.Services.AddTransient<NewTransactionAdapter>();
 builder.Services.AddTransient<LegacyTransactionSystem>();
 builder.Services.AddTransient<NewTransactionSystem>();
+builder.Services.AddTransient<IOrderDeliveryTypeStrategy, InTheEstablishmentOrderStrategy>();
+builder.Services.AddTransient<IOrderDeliveryTypeStrategy, ForTakeawayOrderStrategy>();
```

Comment:
DI couldn't undersand which realization it will be need to use InTheEstablishmentOrderStrategy or ForTakeawayOrderStrategy for interface IOrderDeliveryTypeStrategy

Reply:
![Uploading Screenshot 2024-11-08 at 21.56.47.png…]()

Reply:
https://stackoverflow.com/questions/39174989/how-to-register-multiple-implementations-of-the-same-interface-in-asp-net-core